nixos/modules/security/acme.nix

```diff
@@ -314,21 +314,44 @@ in
                 renewOpts = escapeShellArgs (globalOpts ++
                   [ "renew" "--days" (toString cfg.validMinDays) ] ++
                   certOpts ++ data.extraLegoRenewFlags);
+
+                acmeDnsDeps = optional (data.dnsProvider == "acme-dns")
+                  "acme-dns-${cert}.service";
+
+                commonServiceConfig = {
+                  Type = "oneshot";
+                  User = data.user;
+                  Group = data.group;
+                  PrivateTmp = true;
+                  StateDirectory = "acme/.lego/${cert} acme/.lego/accounts ${lpath}";
+                  StateDirectoryMode = if data.allowKeysForGroup then "750" else "700";
+                  WorkingDirectory = spath;
+                  # Only try loading the credentialsFile if the dns challenge is enabled
+                  EnvironmentFile = if data.dnsProvider != null then data.credentialsFile else null;
+                };
+
                 acmeService = {
                   description = "Renew ACME Certificate for ${cert}";
-                  after = [ "network.target" "network-online.target" ];
+
+                  after = [ "network.target" "network-online.target" ]
+                    ++ acmeDnsDeps;
```

Comment:
This dependency shouldn't be added if acme-dns isn't running on the same machine.